### PR TITLE
ENG-1327/ENG-1441 add to the entandoengine api to check existence of a set of components

### PR DESCRIPTION
**Pull request by w-caffiero-entando**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ampie**

If they are not already assigned, you can assign the PR to them by writing `/assign @ampie` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-engine).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-engine/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 653842578fe188894b3a988d28b37510ab4348ef | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/3.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 89df741d249cc3d85907509679535071773785f5 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/6.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | cfac6dd529349b7e9729048ed2c4eca7ded6cc31 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/7.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 1cdf0f8ce8593b32fe04e438e916b5f6e09cfb9f | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/10.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 7ec2d35e765ba8adbe9dab3d67e14ed0d4eb84f3 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/13.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | a68e6333fc9c5378e83cefa14ac4c95c59df0048 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/14.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 9f697c9da87e98201f42e22d96884a5844747bba | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/15.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 89d030550ea54f8e37e5f23fdaaa0172dc310a05 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/19.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 821778bbf3c7a0adecb4266ae4a1bfaa405e26bc | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/22.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 364acb849c7245b5319bbd8ed44105f40bc95f84 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/23.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | acace81b51bbc4154ea816a1e047e6d18c6c59f8 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/26.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 834ae37f11fc78681bac8afd93f431ac50819bf9 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/27.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 9a755612308a532b69be485adc54c1f07ac71474 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando/entando-engine/PR-81/28.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/main/java/org/entando/entando/aps/system/services/page/PageService.java
Comment (firegloves) [inline src/main/java/org/entando/entando/aps/system/services/page/PageService.java:36]: we should switch to the single file imports

File: src/main/java/org/entando/entando/aps/system/services/page/PageService.java
Comment (firegloves) [inline src/main/java/org/entando/entando/aps/system/services/page/PageService.java:69]: we should switch to the single file imports

File: src/main/java/org/entando/entando/web/analysis/AnalysisController.java
Comment (firegloves) [inline src/main/java/org/entando/entando/web/analysis/AnalysisController.java:57]: remove this empty line?

File: src/main/java/org/entando/entando/web/analysis/IAnalysisController.java
Comment (firegloves) [inline src/main/java/org/entando/entando/web/analysis/IAnalysisController.java:24]: we should switch to the single file imports

**Review by ffleandro — APPROVED** [review]
Looks good to me

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/entando-engine/pull/81#pullrequestreview-533752108" title="Approved">ffleandro</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ampie**

If they are not already assigned, you can assign the PR to them by writing `/assign @ampie` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-engine).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-engine/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
